src/haddock/libs/libpdb.py

```diff
@@ -110,24 +113,64 @@ def get_supported_residues(haddock_topology: FilePath) -> list[str]:
 _to_keep = list(supported_residues)
 
 
-def split_ensemble(pdb_file_path: Path,
-                   dest: Optional[FilePath] = None) -> list[Path]:
+def split_ensemble(
+        pdb_file_path: Path,
+        dest: Optional[FilePath] = None,
+        ) -> list[Path]:
     """
     Split a multimodel PDB file into different structures.
 
     Parameters
     ----------
     dest : str or pathlib.Path
         Destination folder.
+    
+    Returns
+    -------
+    pdb_files_list : list[Path]
+        List of pdb file(s).
     """
     if dest is None:
         dest = Path.cwd()
-    assert pdb_file_path.is_file(), pdb_file_path
+    assert pdb_file_path.is_file(), \
+        f"File '{pdb_file_path}' could not be found in file system."
     with open(pdb_file_path) as input_handler:
         with working_directory(dest):
             split_model(input_handler)
+            pdb_files_list = sort_numbered_paths(
+                *get_new_models(pdb_file_path)
+                )
+    pdb_files_path = [Path(dest, pdb_fname) for pdb_fname in pdb_files_list]
+    return pdb_files_path
+
+
+def count_models(pdb_file_path: FilePath) -> int:
+    """Count number of models in a pdb file.
+    
+    Read filepath and return number of models found in it.
+    If none (not an ensemble), 1 is returned.
 
-    return sort_numbered_paths(*get_new_models(pdb_file_path))
+    Parameters
+    ----------
+    pdb_file_path : FilePath (Union[str, Path])
+        Path to the pdb file to analyse.
+    
+    Returns
+    -------
+    nb_models : int
+        The number of models found in this pdb file.
+    """
+    models_starts: int = 0
+    model_ends: int = 0
+    with open(pdb_file_path, 'r') as filin:
+        for line in filin:
+            if line.startswith("ENDMDL"):
+                model_ends += 1
+            elif line.startswith("MODEL"):
+                models_starts += 1
+    nb_models = max(1, model_ends)
+    assert max(1, models_starts) == nb_models
```

Comment:
Why this assert here?

Reply:
In the case of an ensemble: I am counting `ENDMDL` statements to define the number of models, and I just want to make sure that the number of `ENDMDL` statements is equal to the number of `MODEL` statements.
Otherwise, there should not be any `ENDMDL` nor `MODEL` statements, and I return `1`